habitat_baselines/utils/common.py

```diff
@@ -345,3 +352,78 @@ def overwrite_gym_box_shape(box: Box, shape) -> Box:
     low = box.low if np.isscalar(box.low) else np.min(box.low)
     high = box.high if np.isscalar(box.high) else np.max(box.high)
     return Box(low=low, high=high, shape=shape, dtype=box.dtype)
+
+
+def get_scene_episode_dict(episodes: List[Episode]) -> Dict:
+    scene_ids = []
+    scene_episode_dict = {}
+
+    for episode in episodes:
+        if episode.scene_id not in scene_ids:
+            scene_ids.append(episode.scene_id)
+            scene_episode_dict[episode.scene_id] = [episode]
+        else:
+            scene_episode_dict[episode.scene_id].append(episode)
+
+    return scene_episode_dict
+
+
+def base_plus_ext(path: str) -> Union[Tuple[str, str], Tuple[None, None]]:
+    """Helper method that splits off all extension.
+    Returns base, allext.
+    path: path with extensions
+    returns: path with all extensions removed
+    """
+    match = re.match(r"^((?:.*/|)[^.]+)[.]([^/]*)$", path)
+    if not match:
+        return None, None
+    return match.group(1), match.group(2)
+
+
+def valid_sample(sample: dict) -> bool:
```

Comment:
Doesn't look general enough, possibly better live in IL utils.

Reply:
There's no IL utils so far, and only this function to add. I could add it to the `data.py` file, or we could let this be where it is now.  

This is a webdataset related util function, and might be useful for other IL trainers like `eqa_cnn_pretrain` (even for future trainers) if we were to port that to webdataset in the future for uniformity across all baselines (currently it uses an lmdb dataset).